### PR TITLE
Replace python rt with zig

### PR DESCRIPTION
**Pull request by DrDeano**
Part of #117

### CODE REVIEW

File: src/kernel/log.zig [hunk 1]
```diff
@@ -18,6 +19,20 @@ pub const Level = enum {
 
 var serial: Serial = undefined;
 
+///
+/// The call back function for the std library formate function.
```
Comment (SamTebbs33) [inline src/kernel/log.zig:23]: s/formate/format

File: rt.zig [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+const ChildProcess = std.ChildProcess;
+const Thread = std.Thread;
+const Allocator = std.mem.Allocator;
+const Builder = std.build.Builder;
+const Step = std.build.Step;
+const Queue = std.atomic.Queue([]const u8);
+const Node = std.TailQueue([]const u8).Node;
+
+// Creating a new runtime test:
+// 1. Add a enum to `TestMode`. The name should try to describe the test in one work :P
```
Comment (SamTebbs33) [inline rt.zig:11]: s/work/word

File: src/kernel/arch/x86/isr.zig [hunk 3]
```diff
@@ -188,7 +188,7 @@ fn openIsr(index: u8, handler: idt.InterruptHandler) void {
 ///     IN isr_num: u16 - The isr number to check
 ///
 /// Return: bool
-///     Whether a ISR hander index if valid.
+///     Whether a ISR handler index if valid.
```
Comment (SamTebbs33) [inline src/kernel/arch/x86/isr.zig:191]: s/if/is

File: build.zig [hunk 4]
```diff
@@ -25,96 +28,114 @@ pub fn build(b: *Builder) !void {
 
     const fmt_step = b.addFmt(&[_][]const u8{
         "build.zig",
+        "rt.zig",
```
Comment (SamTebbs33) [inline build.zig:31]: Perhaps rt.zig could be moved to the test dir?

File: build.zig [hunk 5]
```diff
@@ -25,96 +28,114 @@ pub fn build(b: *Builder) !void {
...
         "src",
         "test",
     });
     b.default_step.dependOn(&fmt_step.step);
 
+    comptime var available_tests: []const u8 = "";
```
Comment (SamTebbs33) [inline build.zig:37]: It doesn't look like this is ever used

File: rt.zig [hunk 6]
```diff
@@ -0,0 +1,276 @@
...
+    ///
+    /// Return: bool
+    ///     This will always return true
+    ///
+    fn print_logs(self: *RuntimeStep) bool {
+        while (true) {
+            const msg = self.get_msg() catch return true;
+            defer self.builder.allocator.free(msg);
+        }
+    }
```
Comment (SamTebbs33) [inline rt.zig:99]: This doesn't seem to actually print anything :)

Comment (DrDeano) [reply rt.zig:99]: oops, I forgot to add the warn :P

File: build.zig (hunk 5 above)
Comment (DrDeano) [reply build.zig:37]: Ah, I thing this was the old description for the help, I shall remove

File: build.zig (hunk 4 above)
Comment (DrDeano) [reply build.zig:31]: could do

**Pull request merged**